### DISCUSSION TITLE
Resources Forms Components

### DISCUSSION THREAD

**Discussion by mabdalla123** [filamentphp/filament#4211] [Help] [answered]
how can i disable a component only on Edit Page ?

to solve my problem i  copied visibleOn function  and rename it disabledOn and changed the property to disable and it worked 

is there another way to do the same thing   ?

**Comment by danharrin** [comment, marked as answer]
`->disabled(fn (string $context): bool => $context === 'edit')`

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/forms/advanced#using-closure-customisation

**Reply by mabdalla123** [reply]
Thanks for the help
